src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/PathFactory.groovy

```diff
@@ -0,0 +1,35 @@
+package org.openkilda.functionaltests.helpers.model
+
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.PathNodePayload
+import org.openkilda.northbound.dto.v2.flows.FlowPathV2
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class PathFactory {
+    @Autowired
+    TopologyDefinition topologyDefinition
+
+    Path get(List nodes) {
```

Comment:
My personal preference would be to go strongly typed here and create multiple overloads of 'get' for better IDE and refactoring support. Duck typing does not simplify anything here. But current implementation is 100% fine

Reply:
Having everything strongly typed is an obvious requirement. Unfortunately, Java has problems with overloading generics: JVM considers `void set(List<Integer>)` and `void set(List<String>` as the same signature and raises compliation error.
The workaround could be to use wrapper classes, e.g. List<PathNodePayload> -> PathNodePayloads, but in current code state this would require massive refactoring (e.g. replacing application's DTOs with test's DTOs).